src/main/java/com/google/cloud/tools/app/DeployAction.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app;
+
+import com.google.cloud.tools.app.ProcessCaller.Tool;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Deploys an app to the server.
+ */
+public class DeployAction extends Action {
+
+  private static Set<Option> acceptedFlags = ImmutableSet.of(
+      Option.BUCKET,
+      Option.DOCKER_BUILD,
+      Option.FORCE,
+      Option.IMAGE_URL,
+      Option.PROMOTE,
+      Option.SERVER,
+      Option.VERSION
+  );
+  private File stagedDirectory;
```

Comment:
File.exists(path) I think is the recommended way to do this.

On Thu, Mar 24, 2016 at 3:26 PM, João Martins <notifications@github.com>
wrote:

> In src/main/java/com/google/cloud/tools/app/DeployAction.java
> <https://github.com/GoogleCloudPlatform/app-tools-lib-for-java/pull/1#discussion_r57373968>
> :
>
> > +
> > +/**
> > + * Deploys an app to the server.
> > + */
> > +public class DeployAction extends Action {
> > +
> > +  private static Set<Option> acceptedFlags = ImmutableSet.of(
> > +      Option.BUCKET,
> > +      Option.DOCKER_BUILD,
> > +      Option.FORCE,
> > +      Option.IMAGE_URL,
> > +      Option.PROMOTE,
> > +      Option.SERVER,
> > +      Option.VERSION
> > +  );
> > +  private File stagedDirectory;
>
> Looking at the Path interface, I can't find any "exists()" method. If I
> want to check whether a file exists in the filesystem, I'll have to
> transform the Path into a File and then use exists(). Am I missing anything?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/app-tools-lib-for-java/pull/1/files/6c055eddb488ab5f559df4facacd43818529c737#r57373968>
>